api/remotehastebin.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/jakevoytko/crbot/log"
+)
+
+// RemoteHastebin implements Gist, and interacts with the Hastebin API.
+type RemoteHastebin struct{}
+
+// NewRemoteHastebin works as advertised.
+func NewRemoteHastebin() *RemoteHastebin {
+	return &RemoteHastebin{}
+}
+
+const (
+	MsgHastebinPostFail     = "Unable to connect to Hastebin service. Give it a few minutes and try again"
+	MsgHastebinResponseFail = "Failure reading response from Hastebin service"
+	MsgHastebinStatusCode   = "Failed to upload Hastebin :("
+	MsgHastebinUrlFail      = "Failed getting url from Hastebin service"
+)
+
+func (g *RemoteHastebin) Upload(contents string) (string, error) {
+	response, err := http.Post(
+		"https://hastebin.com/documents", "application/x-www-form-urlencoded", bytes.NewBufferString(contents))
+	if err != nil {
+		log.Info("Error POSTing Hastebin", err)
+		return "", errors.New(MsgHastebinPostFail)
+	} else if response.StatusCode != 200 {
+		log.Info("Bad status code", errors.New("Code: "+strconv.Itoa(response.StatusCode)))
+		body, _ := ioutil.ReadAll(response.Body)
+		log.Info("Response body: ", errors.New(string(body)))
+		return "", errors.New(MsgHastebinStatusCode)
+	}
+	defer response.Body.Close()
+
+	responseMap := map[string]interface{}{}
+	if err := json.NewDecoder(response.Body).Decode(&responseMap); err != nil {
+		log.Info("Error reading Hastebin response", err)
+		return "", errors.New(MsgHastebinResponseFail)
+	}
+
+	if finalUrl, ok := responseMap["key"]; ok {
+		return "https://hastebin.com/" + finalUrl.(string), nil
```

Comment:
Let's link to `/raw/`... the highlighted version highlights a bunch of weird stuff, and this looks a lot better

https://hastebin.com/raw/wemopamusa